Update Post Editor context menu to use new style and update copy

kean opened the pull request:
Fixes https://github.com/wordpress-mobile/WordPress-iOS/issues/11994:

- Update the "More" menu in the editor to use the modern iOS look (easier to reach, can be used with a single gesture, has images to make it easier to find what you need)
- Update some copy to match Gutenberg (they should not invent its own undocumented terminology):
  - "Switch to HTML Mode" → "Code Editor"
  - "Switch to Visual Mode" → "Visual Editor"
  - "History" → "Revisions (%i)"
- Update the copy in the "Revisions" screen to replace "History" with "Revisions"
- Update table view style in the "Revisions" screen to "insetGrouped".

<img width="340" alt="new-context-menu-2" src="https://github.com/wordpress-mobile/WordPress-iOS/assets/1567433/01e9c0ea-58a8-4024-b1a7-dd33e66ef22b">

<img width="340" alt="Screenshot 2024-05-02 at 6 01 52 PM" src="https://github.com/wordpress-mobile/WordPress-iOS/assets/1567433/191e5694-d573-4a9e-98df-e9e3afc424fc">

## To test:

- ✅  Verify the changes according to the acceptance criteria listed in the description

## Known Issues

- ⚠️  "Content Structure" is not immediately available – seems to be an existing issue

## Regression Notes
1. Potential unintended areas of impact: Post Editor
2. What I did to test those areas of impact (or what existing automated tests I relied on): manual
3. What automated tests I added (or what prevented me from doing so): n/a

PR submission checklist:

- [ ] I have completed the Regression Notes.
- [ ] I have considered adding unit tests for my changes.
- [ ] I have considered adding accessibility improvements for my changes.
- [ ] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

Testing checklist:
- [ ] WordPress.com sites and self-hosted Jetpack sites.
- [ ] Portrait and landscape orientations.
- [ ] Light and dark modes.
- [ ] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] VoiceOver.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] iPhone and iPad. 
- [ ] Multi-tasking: Split view and Slide over. (iPad)

kean committed:
Update Post Editor context menu

kean committed:
Update table view style in revisions list

kean committed:
Update copy on the Revisions screen

kean committed:
Update release notes

momo-ozawa reviewed (commented):
Looks great! A few nitpicks:
* Can you remove the separator for the section header in the revisions screen?
* Wdyt of using the same date format for the "post created on ..." footer?

kean committed:
Fix header separator in Revisions list

kean committed:
Update footer

kean commented:
> Can you remove the separator for the section header in the revisions screen?

No idea how I missed it – fixed.

> Wdyt of using the same date format for the "post created on ..." footer? 

I updated the format to `.medium` and also removed it for drafts – as was discussed before, `dateCreated` doesn't mean what it's supposed to mean for drafts. Even on the screenshot you provided, it was incorrect.

<img width="340" alt="Screenshot 2024-05-04 at 11 36 45 AM" src="https://github.com/wordpress-mobile/WordPress-iOS/assets/1567433/32ec6341-3fa3-4ece-a10a-ea841ac07742">
